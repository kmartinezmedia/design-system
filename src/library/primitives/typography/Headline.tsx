import React, { FunctionComponent } from 'react';
import { StyleSheet } from 'react-native';
import { Text, TextProps } from './Text';
import {
  fontFamily,
  fontSize,
  lineHeight,
  fontWeight,
} from '@designSystem/theme';

export const Headline: FunctionComponent<TextProps> = ({
  children,
  spacing = { bottom: 5 },
  ...textProps
}) => (
  <Text style={styles.Headline} spacing={spacing} {...textProps}>
    {children}
  </Text>
);

const styles = StyleSheet.create({
  Headline: {
    fontFamily: fontFamily.medium,
    fontWeight: fontWeight.medium,
    fontSize: fontSize.headline,
    lineHeight: lineHeight.headline,
  },
});
