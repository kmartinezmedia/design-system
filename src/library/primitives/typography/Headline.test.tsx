import React from 'react';
import 'react-native';
import { create, act, ReactTestRenderer } from 'react-test-renderer';
import { Headline } from './Headline';

describe('Headline', () => {
  it('default', () => {
    let component: ReactTestRenderer;
    act(() => {
      component = create(<Headline>Hey, lil 🐛</Headline>);
    });
    expect(component!.toJSON()).toMatchSnapshot();
  });
});
