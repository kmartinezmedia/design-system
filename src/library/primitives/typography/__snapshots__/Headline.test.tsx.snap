// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Headline default 1`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "Graphik-Medium",
        "fontSize": 16,
        "fontWeight": "500",
        "lineHeight": 24,
      },
      Object {
        "paddingBottom": 20,
      },
      Object {
        "color": "#071422",
        "lineHeight": 24,
        "overflow": "visible",
        "textAlign": undefined,
      },
      undefined,
    ]
  }
>
  Hey, lil 🐛
</Text>
`;
